src/Complexes/Show/Gallery.jsx

```diff
@@ -43,38 +46,66 @@ const CounterButton = styled.button`
   border-radius: 0.125rem;
 `;
 
-type Props = { images: Array<GalleryImage>, alt: string };
+export default class GalleryCarousel extends React.Component {
+  state = { isPortalOpened: false, activeImage: 0 };
 
-export default function (props: Props) {
-  const images = props.images || [];
-  const photoPluralize = pluralize(
-    images.length,
-    'фотографий',
-    'фотография',
-    'фотографии',
-    'фотографий',
-  );
+  toggleCarousel = (id: number) => {
+    this.setState({ isPortalOpened: !this.state.isPortalOpened, activeImage: id || 0 });
```

Comment:
https://medium.com/@shopsifter/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1